File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/base/models/LineLengthTest.java
package com.vlad1m1r.lemniscate.base.models;

import com.vlad1m1r.lemniscate.TestConstants;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LineLengthTest {

    private LineLength lineLength;

    @Before
    public void setUp() {
        lineLength = new LineLength();
    }

    @Test
    public void getLineMaxLength() {
        lineLength.setLineMaxLength(0.9f);
        assertEquals(0.9f, lineLength.getLineMaxLength(), TestConstants.DELTA);

    }

    @Test(expected = IllegalArgumentException.class)
    public void getLineMaxLengthException() {
        lineLength.setLineMaxLength(1.1f);
    }

    @Test
    public void setLineMinLength(){
        lineLength.setLineMinLength(0.1f);
        assertEquals(0.1f, lineLength.getLineMinLength(), TestConstants.DELTA);
    }

    @Test(expected = IllegalArgumentException.class)
    public void setLineMinLengthException(){
        lineLength.setLineMaxLength(-1.1f);
    }

}

File: lemniscate/src/test/java/com/vlad1m1r/lemniscate/base/models/LineLengthTest.kt
package com.vlad1m1r.lemniscate.base.models

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class LineLengthTest {

    private val lineLength = LineLength()

    @Test
    fun getLineMaxLength() {
        lineLength.lineMaxLength = 0.9f
        assertThat(lineLength.lineMaxLength).isEqualTo(0.9f)
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMaxLengthGreaterThan1ThrowsException() {
        lineLength.lineMaxLength = 1.1f
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMaxLengthEqualTo0ThrowsException() {
        lineLength.lineMaxLength = 0.0f
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMaxLengthLestThan0ThrowsException() {
        lineLength.lineMaxLength = -1.0f
    }

    @Test
    fun setLineMinLength() {
        lineLength.lineMinLength = 0.1f
        assertThat(lineLength.lineMinLength).isEqualTo(0.1f)
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMinLengthLestThan0ThrowsException() {
        lineLength.lineMinLength = -0.1f
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMinLengthGreaterThan1ThrowsException() {
        lineLength.lineMinLength = 1.1f
    }

    @Test(expected = IllegalArgumentException::class)
    fun setLineMinLengthEqualTo0ThrowsException() {
        lineLength.lineMinLength = 0.0f
    }
}